src/lib/server/ApiClient.ts

```diff
@@ -0,0 +1,26 @@
+export type UploadRequest = {
+    ObjectId: number,
+    ObjectKey: string,
+    StatusName: string
+}
+
+export async function getUploadRequest(): Promise<UploadRequest[]> {
+ 
+    try{
+        const uploadRequest: UploadRequest[] =  await fetch('https://gx9mczg2il.execute-api.us-east-1.amazonaws.com/api/v1/users/1/upload/requests')
+                                            .then(function(response) {
+                                                if(!response.ok){
+                                                    throw response.statusText;
+                                                }
+                                                return response.json();
+                                            });
+
+  
+
+        return uploadRequest;
+    }
+    catch(e){
+        console.error(e);
+        return null;
+    }
```

Comment:
What is the point of returning `null`? How is a user of the function should know what happened?
There can be a lot of different errors, such as 503 (server went down), 404 (sometimes it is ok to have this error), 403, 401 (auth related).

The point is that API client can't handle all errors by itself. E.g. if a client doesn't have an internet connection we should show a popup. When you return just `null` it is not possible.

Some errors can be handled by API client. E.g. 503 (service temporary unavailable). API client can perform retries, but if failed, the client of a function also should know about an error.

In conclusion, it is important to outline who's responsible for handling particular errors.